### PR TITLE
Add Cron to Repo indexer

### PR DESCRIPTION
**Pull request by zheng1**

### CODE REVIEW

File: pkg/manager/repo_indexer/cron.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		if err != nil {
+			return nil, err
+		}
+		for _, r := range res.GetRepoSet() {
+			rs = append(rs, &repoInfo{
+				repoId: r.GetRepoId().GetValue(),
+				owner:  r.GetOwner().GetValue(),
+			})
+		}
+		if len(res.GetRepoSet()) >= int(limit) {
```
Comment (chilianyi) [inline pkg/manager/repo_indexer/cron.go]: If length is bigger than 50, the item has already been appended into rs.

File: pkg/manager/repo_indexer/cron.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+func (p *Server) startCron(repoCron string) *cron.Cron {
+	c := cron.New()
+	if repoCron != "" {
+		c.AddFunc(repoCron, func() {
+			logger.Debugf("Start auto index, current cron is [%s]", repoCron)
+			err := p.autoIndex()
+			if err != nil {
+				logger.Panicf("failed to auth index repoInfos, [%+v]", err)
```
Comment (chilianyi) [inline pkg/manager/repo_indexer/cron.go]: auto

File: pkg/manager/repo_indexer/cron.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	logger.Debugf("Repo cron had started")
+	return c
+}
+
+func (p *Server) Cron() {
+	repoCron := p.GlobalConfig().Repo.Cron
+	c := p.startCron(repoCron)
+	p.ThreadWatchGlobalConfig(func(globalConfig *config.GlobalConfig) {
+		currentRepoCron := globalConfig.Repo.Cron
+		if currentRepoCron != repoCron {
```
Comment (chilianyi) [inline pkg/manager/repo_indexer/cron.go]: Will repoCron be stored?

File: pkg/manager/repo_indexer/cron.go (hunk 1 above)
Comment (zheng1) [reply pkg/manager/repo_indexer/cron.go]: Yes, this func will return all of the repo info.

File: pkg/manager/repo_indexer/cron.go (hunk 3 above)
Comment (zheng1) [reply pkg/manager/repo_indexer/cron.go]: The lasted version will store at etcd.

File: pkg/manager/repo_indexer/cron.go (hunk 2 above)
Comment (zheng1) [reply pkg/manager/repo_indexer/cron.go]: done

**Pull request merged**